.github/workflows/build_and_test.yml

```diff
@@ -90,6 +90,28 @@ jobs:
         if: ${{ matrix.bazel-command == 'dazel' }}
         run: ${{ matrix.sudo-command }} pip3 install dazel
 
+      # Stop any running Dazel instances, so that any volume mounts for the
+      # caches removed above don't stick around. Volume mounts for nonexistent
+      # directories cause build failures.
+      #
+      # Note that after a clean reboot this may print an error saying that
+      # `docker stop` was given no arguments, as there were no containers running
+      # at the time: that's ok, so pipe errors to /dev/null.
+      - name: Stop Dazel instances
+        if: ${{ matrix.bazel-command == 'dazel' }}
+        run: |
+          docker ps --filter name="dazel/*" --filter status=running -aq \
+            | xargs docker stop 2> /dev/null \
+            || true
+
+      # Clean up any artifacts from previous builds to ensure that we aren't
+      # mixing state across novel and old builds. This is especially important
+      # if new builds use different compiler settings than old builds.
+      - name: Remove Dazel cache
+        if: ${{ matrix.bazel-command == 'dazel' }}
+        run: |
+          sudo rm -rf ~/.cache/dazel
+
```

Comment:
I'm surprised that this is necessary. This is the `eventuals` repository, and it uses only GitHub-hosted Action Runners. Those should be hermetic. Why is it necessary to stop the Dazel instances and remove the cache manually?

Reply:
Excellent question! It's not necessary at all: I tried several magic incantations and failed to isolate what the fix actually is.

You're right, the cleanup is unnecessary because each actions runner should run in its own VM (per some reading I did, though I didn't find a super strong conclusion on this).

What actually matters is removing the `--spawn_strategy=local` bazel flag. Per https://bazel.build/docs/user-manual#execution-strategy , the `local` flag runs compilation steps as a local subprocess (thereby accessing whatever files / libraries are available locally). The default when not specifying a flag is to instead run in some sort of sandbox which restricts where it can look for libraries. I'm not sure why we were previously preferring the local spawn strategy, and since removing it fixes the build let's do that :)

Reply:
> the local flag runs compilation steps as a local subprocess (thereby accessing whatever files / libraries are available locally). The default when not specifying a flag is to instead run in some sort of sandbox which restricts where it can look for libraries.

Huh? That doesn't sound right. There's no way it just stops accessing the local libraries. There should be something else.